File: netmapgenerator/PngClustering/ImageTransform/Node.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ImageTransform
{
    public class Node
    {
        public int X { get; set; }
        public int Y { get; set; }
        public Dictionary<Node, int> Pings { get; set; }
        public List<Node> SortedPings { get; set; }
        public bool Active { get; set; }

        public Node()
        {
            Pings = new Dictionary<Node, int>();
        }

        public double Distance(Node node)
        {
            return Math.Sqrt(Math.Pow(node.X - X, 2) + Math.Pow(node.Y - Y, 2));
        }

        public override string ToString()
        {
            return "X " + X + " Y" + Y;
        }
    }

    public class PingNode
    {
        public Node Node { get; set; }
        public int Ping { get; set; }
    }
}

File: generator/algorithm/node.go
package algorithm

import (
	"fmt"
	"math"
)

type EdgeNode struct {
	Name string  `json:"name"`
	X    float64 `json:"x"`
	Y    float64 `json:"y"`
	//SortedPings []*FogNode
	//Pings       []float32
	Pings map[string]float32 `json:"pings"` //remove the map!!!
}

func (en *EdgeNode) Init() *EdgeNode {
	en.Pings = make(map[string]float32)
	return en
}

func (en *EdgeNode) Distance(other *FogNode) float64 {
	return math.Sqrt(math.Pow(en.X-other.X, 2) + math.Pow(en.Y-other.Y, 2))
}

func (en *EdgeNode) String() string {
	return fmt.Sprintf("X%f Y%f", en.X, en.Y)
}

type FogNode struct {
	X     float64            `json:"x"`
	Y     float64            `json:"y"`
	Name  string             `json:"name"`
	Pings map[string]float32 `json:"pings"`
	//	Active         bool
	//	ResourcesUsed  map[Resource]int
	//	ResourcesTotal map[Resource]int
}

func (fn *FogNode) Init() *FogNode {
	//	fn.ResourcesTotal = make(map[Resource]int)
	//	fn.ResourcesUsed = make(map[Resource]int)
	return fn
}

func (fn *FogNode) String() string {
	return fmt.Sprintf("X%d Y%d", fn.X, fn.Y)
}

/*func (fn *FogNode) IsFull() bool {
	full := false

	for res, amount := range fn.ResourcesUsed {
		if amount >= fn.ResourcesTotal[res] {
			full = true
		}
	}

	return full
}*/

/*func (fn *FogNode) IsUnderutilized() bool {
	underfull := true

	for res, amount := range fn.ResourcesUsed {
		if float64(amount) >= 0.2*float64(fn.ResourcesTotal[res]) {
			underfull = false
		}
	}

	return underfull
}*/

/*func (fn *EdgeNode) GetPing(fNode *FogNode) float32 {
	idx := 0
	var ping float32
	found := false
	for idx < len(fn.SortedPings) && !found {
		if fn.SortedPings[idx] == fNode {
			ping = fn.Pings[idx]
			found = true
		}

		idx++
	}
	return ping
}*/
